### PR TITLE
Feature/rtodling/v1 5 4 onl xoffl flags

### PR DESCRIPTION
**Pull request by rtodling**
We've had the issue of having mixed variable definition between what's to be handled at setup and what's to be handled at run time for quite sometime. The bottom line is this: 

variables labeled @SOMETHING are setup variables
variable labeled >>>SOMETHING<<< are run-time vars

The DAS does not mix them, but the GCM unfortunately does. The present changes makes a minor contribution toward adjusting these on the GCM side for two variables:

COUPLED
DATAOCEAN

The changes here also add a variable CLDMICRO that defines the micro physics opt used - defaults to 1MOMENT -  and only in COUPLED mode it is redefined as 2MOMENT; that is, this is no longer wired in the COUPLED section of the AGCM.rc.tmpl - since we want to ability to decide the micro physic opt at setup time for the couple case.

Notice that there seems to be already a conflict - not introduced by these changes - in the setup  where there seems to be two ways of controlling the microphysics options: 

a) one is controlled by the variable CLDMICRO of the AGCM.rc.tmpl - not to be confused w/ the script variable I introduced of the same name (for consistency) --- this option is triggered when setting up a COUPLED exp.
b) and another called: MP_MG1CLDMICRO - which is present in both AMIP and COUPLED modes

How (in)consistent these are in the setup and how these interact in the model is beyond me.

In any case, the changes here to do fuss w/ trying to address spotted inconsistencies. I simply coordinates var names between the model and DAS setups. All changes should be zero diff and have the intent to allow the ADAS to setup a coupled run optio,  which it now does.

### CODE REVIEW

File: AGCM.rc.tmpl [hunk 1]
```diff
@@ -658,17 +658,17 @@ SEAICETHERMO_INTERNAL_RESTART_FILE:     seaicethermo_internal_rst
 SEAICETHERMO_INTERNAL_CHECKPOINT_FILE:  seaicethermo_internal_checkpoint
 SEAICETHERMO_INTERNAL_CHECKPOINT_TYPE:  @CHECKPOINT_TYPE
 
->>>COUPLED<<<CICE4_IMPORT_RESTART_FILE:             seaice_import_rst
->>>COUPLED<<<CICE4_IMPORT_CHECKPOINT_FILE:          seaice_import_checkpoint
->>>COUPLED<<<CICE4_IMPORT_CHECKPOINT_TYPE:          @CHECKPOINT_TYPE
+CICE4_IMPORT_RESTART_FILE:             seaice_import_rst
```
Comment (mathomp4) [inline AGCM.rc.tmpl:661]: @rtodling These might need to be `@COUPLED` protected.

Comment (sanAkel) [reply AGCM.rc.tmpl:661]: Does "@a" work the same way as ">>>" If so I *prefer former* - it is less stuff to type and look at!

File: AGCM.rc.tmpl [hunk 2]
```diff
@@ -815,22 +815,22 @@ USE_SATSIM_MISR:  @MISR_SATSIM
 >>>MOM5<<<#DATA_SST_FILE: sst.data
```
Comment (sanAkel) [inline AGCM.rc.tmpl:815]: @MOM5 for these two lines

File: gcm_forecast.tmpl [hunk 3]
```diff
@@ -392,43 +392,43 @@ done:
...
 setenv CHMDIR    @CHMDIR
->>>DATAOCEAN<<<setenv BCRSLV    @ATMOStag_@OCEANtag
->>>COUPLED<<<setenv BCRSLV    @ATMOStag_DE0360xPE0180
+@DATAOCEAN setenv BCRSLV    @ATMOStag_@OCEANtag
+@COUPLED setenv BCRSLV    @ATMOStag_DE0360xPE0180
 setenv DATELINE  DC
 setenv EMISSIONS @EMISSIONS
 
 >>>MOM5<<<setenv GRIDDIR  @COUPLEDIR/a${AGCM_IM}x${AGCM_JM}_o${OGCM_IM}x${OGCM_JM}
 >>>MOM6<<<setenv GRIDDIR  @COUPLEDIR/MOM6/@ATMOStag_@OCEANtag
```
Comment (sanAkel) [inline gcm_forecast.tmpl:403]: @MOM5 *and* @MOM6

File: gcm_run.j [hunk 4]
```diff
@@ -300,43 +300,43 @@ done:
...
->>>COUPLED<<</bin/ln -sf $GRIDDIR/@ATMOStag_@OCEANtag-Pfafstetter.til   tile.data
->>>COUPLED<<</bin/ln -sf $GRIDDIR/@ATMOStag_@OCEANtag-Pfafstetter.TRN   runoff.bin
->>>COUPLED<<</bin/ln -sf $GRIDDIR/MAPL_Tripolar.nc .
->>>COUPLED<<</bin/ln -sf $GRIDDIR/vgrid${OGCM_LM}.ascii ./vgrid.ascii
+@COUPLED /bin/ln -sf $GRIDDIR/SEAWIFS_KPAR_mon_clim.${OGCM_IM}x${OGCM_JM} SEAWIFS_KPAR_mon_clim.data
+@COUPLED /bin/ln -sf $GRIDDIR/@ATMOStag_@OCEANtag-Pfafstetter.til   tile.data
+@COUPLED /bin/ln -sf $GRIDDIR/@ATMOStag_@OCEANtag-Pfafstetter.TRN   runoff.bin
+@COUPLED /bin/ln -sf $GRIDDIR/MAPL_Tripolar.nc .
+@COUPLED /bin/ln -sf $GRIDDIR/vgrid${OGCM_LM}.ascii ./vgrid.ascii
 >>>MOM5<<</bin/ln -s @COUPLEDIR/a@HIST_IMx@HIST_JM_o${OGCM_IM}x${OGCM_JM}/DC0@HIST_IMxPC0@HIST_JM_@OCEANtag-Pfafstetter.til tile_hist.data
```
Comment (sanAkel) [inline gcm_run.j:330]: @MOM5

File: gcm_setup [hunk 5]
```diff
@@ -2097,10 +2098,11 @@ s?@DASTUNING?#?g
 
 s?>>>FORCEDAS<<<?$FORCEDAS?g
 s?>>>FORCEGCM<<<?$FORCEGCM?g
-s?>>>COUPLED<<<?$COUPLED?g
+s?@COUPLED?$COUPLED?g
+s?@CLDMICRO?$CLDMICRO?g
 s?>>>MOM5<<<?$MOM5?g
```
Comment (sanAkel) [inline gcm_setup:2103]: @MOM5 and @MOM6

File: gcm_setup [hunk 6]
```diff
@@ -2097,10 +2098,11 @@ s?@DASTUNING?#?g
 
 s?>>>FORCEDAS<<<?$FORCEDAS?g
 s?>>>FORCEGCM<<<?$FORCEGCM?g
-s?>>>COUPLED<<<?$COUPLED?g
+s?@COUPLED?$COUPLED?g
```
Comment (sanAkel) [inline gcm_setup:2101]: @COUPLED

**Review by sanAkel — CHANGES_REQUESTED** [review]
needs testing ...

File: AGCM.rc.tmpl (hunk 1 above)
Comment (mathomp4) [reply AGCM.rc.tmpl:661]: My main question was is it okay to have these not be protected by `@COUPLED`. I suppose it is, I mean, other restarts are exposed at all times...

File: gcm_run.j [hunk 7]
```diff
@@ -913,18 +913,18 @@ foreach collection ( $collections )
    /bin/mv `/bin/ls -1 *.${collection}.*` $EXPDIR/holding/$collection
 end
 
->>>COUPLED<<<# MOM-Specific Output Files
->>>COUPLED<<<# -------------------------
+@COUPLED # MOM-Specific Output Files
+@COUPLED # -------------------------
 >>>MOM5<<< set dsets="ocean_month"
 >>>MOM6<<< set dsets="ocean_state prog_z sfc_ave forcing"
```
[the remark this answers is not in the document]
Comment (yvikhlya) [reply gcm_run.j:919]: > What if I prefer to call `prog_z` in _my version_ of `diag_table` `prog_z_WOA`?

Then you need to make a suggestion, how gcm_run script is supposed to know which files to copy.

[the remark this answers is not in the document]
Comment (sanAkel) [reply gcm_run.j:919]: something to the following effect:

set `mom_output_file_names = x1, x2, x3, ...` does that work for you? Please, *again* in a *LATER* PR. Thanks!

**Review by mathomp4 — APPROVED** [review]
Tested with AMIP, MOM5, MOM6. All seemed to run and were zero-diff.

**Review by sanAkel — APPROVED** [review]
All requested changes seem to have been made. We have to test the new `1440x1080` configuration.

**Pull request merged**